src/redisearch_rs/inverted_index/src/tests.rs

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use std::mem::ManuallyDrop;
+
+use ffi::t_docId;
+
+use crate::{RSAggregateResult, RSIndexResult, RSIndexResultData, RSResultType, RSResultTypeMask};
+
+#[unsafe(no_mangle)]
+pub extern "C" fn IndexResult_ConcatMetrics(
+    _parent: *mut RSIndexResult,
+    _child: *mut RSIndexResult,
+) {
+    // Do nothing since the code will call this
+}
+
+impl RSAggregateResult {
+    fn get(&self, index: usize) -> Option<&RSIndexResult> {
+        if index > self.num_children as _ {
+            return None;
+        }
+
+        let child_ptr = unsafe { *self.children.add(index) };
+
+        if child_ptr.is_null() {
+            None
+        } else {
+            let child = unsafe { &*child_ptr };
+            Some(child)
+        }
+    }
+}
+
+impl RSIndexResult {
+    fn union(doc_id: t_docId, cap: usize, weight: f64) -> Self {
+        Self {
+            doc_id,
+            dmd: std::ptr::null(),
+            field_mask: 0,
+            freq: 0,
+            offsets_sz: 0,
+            data: RSIndexResultData {
+                agg: ManuallyDrop::new(RSAggregateResult::new(cap)),
+            },
+            result_type: RSResultType::Union,
+            is_copy: false,
+            metrics: std::ptr::null_mut(),
+            weight,
+        }
+    }
+
+    fn get(&self, index: usize) -> Option<&RSIndexResult> {
+        if self.is_aggregate() {
+            let agg = unsafe { &self.data.agg };
+
+            agg.get(index)
+        } else {
+            None
+        }
+    }
+}
+
+#[test]
+fn pushing_to_aggregate_result() {
```

Comment:
Did you check with `valgrind` if we don't leak memory in those tests?
I find it useful when writing code manually allocating memory.

Reply:
Ahhh, I see. I'm purposefully not cleaning up the children pointers because that only has to happen if the type was copied (deep cloned). However, I should be able to have the tests clean up the memory correctly.

I'll add an update for this

Reply:
Done in https://github.com/RediSearch/RediSearch/pull/6490/commits/7c5c48d81ca0acdb075d93bcdeaa59ef7bbc7721